integration-cli/docker_cli_pull_test.go

```diff
@@ -150,3 +150,34 @@ func (s *DockerSuite) TestPullScratchNotAllowed(c *check.C) {
 		c.Fatalf("unexpected output pulling scratch: %s", out)
 	}
 }
+
+// pulling an image with --all-tags=true
+func (s *DockerSuite) TestPullImageWithAllTagFromCentralRegistry(c *check.C) {
+	testRequires(c, Network)
+	pullCmd := exec.Command(dockerBinary, "pull", "busybox")
+	if out, _, err := runCommandWithOutput(pullCmd); err != nil {
+		c.Fatalf("pulling the busybox image from the registry has failed: %s, %v", out, err)
+	}
+
+	ImageCmd := exec.Command(dockerBinary, "images", "busybox")
+	outImageCmd, exitCode, err := runCommandWithOutput(ImageCmd)
+
+	if err != nil || exitCode != 0 {
```

Comment:
If err is not nil exit code will be not equal to zero so you can omit the second check, you can just use `c.Assert(err, check.IsNil)`